spec/puppet-lint/plugins/puppet-lint_duplicate_class_parameters_spec.rb

```diff
@@ -162,4 +162,22 @@ class duplicated_type(
     end
   end
 
+  context 'class with a structure using a variable and assigned in another variable' do
+    let(:code) do
+      <<-EOS
+        class complex_structure_assignation (
+          Optinal[String] $hostname = undef,
```

Comment:
Should this be `Optional` ?

Reply:
Yes indeed!  I amended the commit to fix the issue.